spec/util.spec.js

```diff
@@ -88,6 +88,171 @@ describe('util.ajax:', function() {
+
+            done();
+        });
+    });
+
+    it('load text response', function(done) {
+        let options = {
+            url: '/base/spec/support/test.txt',
+            responseType: 'text'
+        };
+        let instance = WaveSurfer.util.fetchFile(options);
+        instance.once('success', data => {
+            expect(instance.response.status).toEqual(200);
+            expect(instance.response.headers.get('content-type')).toEqual(
+                'text/plain'
+            );
+
+            // returned data is a string
+            expect(data).toEqual('hello world');
+
+            done();
+        });
+    });
+
+    it('load unknown response type', function(done) {
+        let options = {
+            url: audioExampleUrl,
+            responseType: 'fooBar'
+        };
+        let instance = WaveSurfer.util.fetchFile(options);
+        instance.once('error', error => {
+            expect(error).toEqual(
+                new Error('Unknown responseType: ' + options.responseType)
+            );
+
+            done();
+        });
+    });
+
+    it('throws error when URL contains credentials', function() {
+        let options = {
+            url: 'http://user:password@example.com'
+        };
+        try {
+            WaveSurfer.util.fetchFile(options);
+        } catch (err) {
+            expect(err).toEqual(jasmine.any(TypeError));
+        }
+    });
+
+    it('throws error when URL is missing', function() {
+        try {
+            WaveSurfer.util.fetchFile({});
+        } catch (err) {
+            expect(err).toEqual(new Error('fetch url missing'));
+        }
+    });
+
+    it('throws error when options are missing', function() {
+        try {
+            WaveSurfer.util.fetchFile();
+        } catch (err) {
+            expect(err).toEqual(new Error('fetch options missing'));
+        }
+    });
+
+    it('fires error event when the file is not found', function(done) {
+        let options = {
+            url: '/foo/bar'
+        };
+        let instance = WaveSurfer.util.fetchFile(options);
+        instance.once('error', error => {
+            expect(instance.response.status).toEqual(404);
+            expect(error).toEqual(new Error('HTTP error status: 404'));
+
+            done();
+        });
+    });
+
+    it('accepts custom request headers', function(done) {
+        let options = {
+            url: '/base/spec/support/test.txt',
+            responseType: 'text',
+            requestHeaders: [
+                {
+                    key: 'Content-Type',
+                    value: 'text/plain'
+                }
+            ]
+        };
+        let instance = WaveSurfer.util.fetchFile(options);
+        instance.once('success', data => {
+            expect(instance.response.headers.has('Content-Type')).toBeTrue();
+            expect(instance.response.headers.get('Content-Type')).toEqual(
+                'text/plain'
+            );
+
+            done();
+        });
+    });
+});
+
```

Comment:
All of the `let`s in this file can be changed to `const` too. Not a big deal though.